manifests/security.pp

```diff
@@ -17,17 +17,91 @@
 # Jenkins security configuration
 #
 class jenkins::security (
-  $security_model = undef,
+  $jenkins_management_email         = '',
+  $jenkins_management_login         = '',
+  $jenkins_management_name          = '',
+  $jenkins_management_password      = '',
+  $jenkins_s2m_acl                  = false,
+  $jenkins_ssh_private_key_contents = '',
+  $jenkins_ssh_public_key_contents  = '',
+  $ldap_access_group                = '',
+  $ldap_group_search_base           = '',
+  $ldap_inhibit_root_dn             = 'no',
+  $ldap_manager                     = '',
+  $ldap_manager_passwd              = '',
+  $ldap_overwrite_permissions       = '',
+  $ldap_root_dn                     = 'dc=company,dc=net',
+  $ldap_uri                         = 'ldap://ldap',
+  $ldap_user_search                 = 'uid={0}',
+  $ldap_user_search_base            = '',
+  $security_model                   = undef,
 ){
   validate_string($security_model)
 
   include ::jenkins::cli_helper
 
+  $libdir = $::jenkins::libdir
+
+  file { "${libdir}/.ssh/id_rsa" :
+    owner   => 'jenkins',
+    group   => 'jenkins',
+    mode    => '0600',
+    content => $jenkins_ssh_private_key_contents,
+    replace => true,
+    require => File["${libdir}/.ssh/"],
+  }
+
+  file { "${libdir}/.ssh/id_rsa.pub" :
+    owner   => 'jenkins',
+    group   => 'jenkins',
+    mode    => '0644',
+    content => "${jenkins_ssh_public_key_contents} jenkins@${::fqdn}",
+    replace => true,
+    require => File["${libdir}/.ssh"],
+  }
+
+  if ($security_model == 'unsecured') {
+    $security_opt_params = 'set_security_unsecured'
+  }
+
+  if ($security_model == 'ldap') {
```

Comment:
One suggestion, since this is a lot to place on the command line. What about writing a configuration file to disk temporarily and then having the groovy helper read that configuration file?

I worry that there command line might end up being too long

Reply:
@matonb That code has been merged into this module.